WIP Command Streams

cindykimxp opened the pull request:
**Fixes issue:**
 #351, the issues presented in runlib.in-toto-run

**Description of the changes being introduced by the pull request:**
- To correctly print streams to console or record streams in link metadata at appropriate times
- Add quiet keyword argument (default True) to enable (False) printing streams to console
- Use record_streams keyword (default False) to enable (True) recording streams in link metadata

Follows the logic:
When record_streams is true:
- if quiet is true --> link metadata
- if quiet is false --> both link metadata and terminal

When record_streams is false:
- if quiet is true --> nowhere
- if quiet is false --> terminal
 
Note: the previous PR referencing this issue was closed because of a complicated commit/git history -- this was causing problems with DCO sign-off and other rebasing issues. 

**Please verify and check that the pull request fulfills the following
 requirements:**
- The code follows the Code Style Guidelines
- Tests have been added for the bug fix or new feature
- Docs have been added for the bug fix or new feature

cindykimxp committed:
Update runlib.py

Signed-off-by: cindykimxp <cindykim100@gmail.com>

lukpueh commented:
Thanks for untangling the git history of #357. Before we move forward, please make sure that
- all comments from my last review in https://github.com/in-toto/in-toto/pull/357#pullrequestreview-459896874 are addressed,
- your new code is tests,
- and @kristelfung's new timeout setting (https://github.com/in-toto/in-toto/pull/367) is respected in all subprocess calls.

cindykimxp committed:
Argment Description and Unit Test

Added description of argument quiet to docstrings of execute_link and in_toto_run. Also created unit test for the function of quiet when record_streams is true and quiet is false.

Signed-off-by: cindykimxp <cindykim100@gmail.com>

cindykimxp committed:
Trailing Whitespace Fix and Conditions Fix

Fixed the trailing whitespace following the unit test, and making sure to close the buffer. Fixed minor error in docstring for quiet.

Signed-off-by: cindykimxp <cindykim100@gmail.com>

cindykimxp committed:
Whitespace Lines

Cleared lines with trailing whitespace.

Signed-off-by: cindykimxp <cindykim100@gmail.com>

cindykimxp committed:
Cleared in runlib.py

Cleared remaining whitespaces in runlib.py

Signed-off-by: cindykimxp <cindykim100@gmail.com>

cindykimxp committed:
Exception Type

Specifying exception type as ImportError (StringIO may not be accepted depending on Python version)

Signed-off-by: cindykimxp <cindykim100@gmail.com>

cindykimxp committed:
Removed except as

Cleaned up import except condition

Signed-off-by: cindykimxp <cindykim100@gmail.com>

cindykimxp committed:
.DS_Store banished!

Signed-off-by: cindykimxp <cindykim100@gmail.com>

cindykimxp committed:
Update test_runlib.py

- Redirecting stdout
- Correctly saves nothing into record_streams (since it's False)
- Does NOT work for quiet (f.getvalue() is nothing when it should be 'hello')

Signed-off-by: cindykimxp <cindykim100@gmail.com>

adityasaky commented:
Closing this PR now, thanks for your work @cindykimxp.
